From 43870c99b1baa727bb1071903036ee9c395d0a11 Mon Sep 17 00:00:00 2001
From: Fabien Brisset <fabien.brisset@alcatel-lucent.com>
Date: Tue, 24 Apr 2012 13:30:26 +0200
Subject: [PATCH 1/8] Adding Ethernet support

---
 core/pathmap.mk |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/core/pathmap.mk b/core/pathmap.mk
index 60fbfd2..5cfa5d9 100644
--- a/core/pathmap.mk
+++ b/core/pathmap.mk
@@ -87,6 +87,7 @@ FRAMEWORKS_BASE_SUBDIRS := \
 	    opengl \
 	    sax \
 	    telephony \
+	    ethernet \
 	    wifi \
 	    keystore \
 	    icu4j \
-- 
1.7.4.1

