From f382ef9e6bc7c00c20bfb49a8ecd9b95f47f2ff8 Mon Sep 17 00:00:00 2001
From: Fabien Brisset <fabien.brisset@alcatel-lucent.com>
Date: Thu, 21 Jun 2012 17:20:21 +0200
Subject: [PATCH 13/50] Adding missing Ethernet support in ConnectivityService

---
 .../com/android/server/ConnectivityService.java    |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/services/java/com/android/server/ConnectivityService.java b/services/java/com/android/server/ConnectivityService.java
index 002fb56..8e009b8 100644
--- a/services/java/com/android/server/ConnectivityService.java
+++ b/services/java/com/android/server/ConnectivityService.java
@@ -2877,6 +2877,8 @@ private NetworkStateTracker makeWimaxStateTracker() {
             } else {
                 Slog.e(TAG, "Can't match any wifi netTracker!");
             }
+        } else if (networkType == ConnectivityManager.TYPE_ETHERNET) {
+           usedNetworkType = ConnectivityManager.TYPE_ETHERNET;
         } else {
             Slog.e(TAG, "Unexpected network type");
         }
-- 
1.7.4.1

